### DISCUSSION TITLE
using with Blazor Desktop

### DISCUSSION THREAD

**Discussion by BenMakesGames** [Blazored/Modal#339] [Q&A] [answered]
are there any special considerations for using Blazored.Modal within a Blazor Desktop app? it's not working for me (no errors or apparent changes in the DOM after calling `.Show<...>()`), but Blazor Desktop is the first time I've used Blazor at all (though I have ASP.NET MVC and Angular experience), so adapting "normal" Blazor instructions to Blazor Desktop does not yet come naturally to me.

---

in case it's helpful...

setting up app builder:
```cs
        public App()
        {
            var builder = MobileBlazorBindingsHost.CreateDefaultBuilder()
                .ConfigureAppConfiguration(ConfigureConfiguration)
                .ConfigureLogging(ConfigureLogging)
                .ConfigureServices(ConfigureServices)
                .UseWebRoot("wwwroot")
                .UseStaticFiles()
            ;
```

ConfigureServices method (called from above):
```cs
        private void ConfigureServices(HostBuilderContext context, IServiceCollection services)
        {
            services.AddBlazorHybrid();

            // singleton services
            services.AddSingleton(new Random());
            services.AddBlazoredModal();
```

index.html:
```html
    <head>
        <meta charset="utf-8"/>
        <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no"/>
        <base href="/"/>
        <link href="_content/Blazored.Modal/blazored-modal.css" rel="stylesheet"/>
        <link href="_content/DesktopUI/css/main.css" rel="stylesheet"/>
        <script src="_content/Blazor.Extensions.Canvas/blazor.extensions.canvas.js"></script>
        <script src="_content/Blazored.Modal/blazored.modal.js"></script>
    </head>
```

a component:
```cs
        [Inject] private IModalService Modals { get; set; }

        ...

        private void DoConfirmDelete()
        {
            Modals.Show<ConfirmDeleteDialog>();
        }
```

[edit] almost forgot: the main.razor:
```razor
@using Microsoft.AspNetCore.Components.Routing;
<ContentView>
    <StackLayout>
        <BlazorWebView VerticalOptions="LayoutOptions.FillAndExpand">
            <CascadingBlazoredModal>
                <Router AppAssembly="@GetType().Assembly">
                    <Found Context="routeData">
                        <RouteView RouteData="@routeData"/>
                    </Found>
                    <NotFound>
                        <p>Sorry, there's nothing at this address.</p>
                    </NotFound>
                </Router>
            </CascadingBlazoredModal>
        </BlazorWebView>
    </StackLayout>
</ContentView>
```

**Comment by larsk2009** [comment]
Hi @BenMakesGames,

I am afraid I have no experience (yet) with Blazor Desktop. Maybe @chrissainty does, otherwise could you share an (example) project where you see this issue? Then I can open it in Visual Studio and see if I can figure out the issue.

**Comment by chrissainty** [comment, marked as answer]
I'm in the same boat as @larsk2009 here. I've not had any chance to play with Blazor Desktop.

**Reply by BenMakesGames** [reply]
no problem: I eventually found the issue: I was just missing a `using` statement, and that wasn't preventing it from building (surprising!) I had left all the Blazored.Modal code in, expecting the issue would be fixed with the release of .NET 6, and happened, just yesterday, to notice a build _warning_ message which hinted at the problem.

I'm pretty new to Blazor in general... sorry for the noobie mistake, and sorry for wasting your time!

now that Blazored.Modal is working, I can at least say that I love it :P

**Reply by BenMakesGames** [reply]
a thought: if there's a way that calling `.Show(...)` could detect that it doesn't have a CascadingBlazoredModal, and throw an exception, rather than silently fail, that'd be cool... or maybe throw that exception during the initialization of the IModalService? I mean, ideally the build would fail if you use a tag that hasn't been properly included, but I guess we can't fix that :P

**Reply by chrissainty** [reply]
That's what this code should do:

https://github.com/Blazored/Modal/blob/e3a6c78f12440caf27c8b06342f61f842f397795/src/Blazored.Modal/BlazoredModal.razor.cs#L36-L39
